api/src/main/java/org/apache/iceberg/util/CharSequenceSet.java

```diff
@@ -172,7 +172,7 @@ public boolean equals(Object o) {
 
   @Override
   public int hashCode() {
-    return Objects.hash(wrapperSet);
```

Comment:
1.) Yeah I was thinking about this. It would change the value of the hashcode compared to using hash, because before we're doing an array allocation and then hashing that, now we're doing the hash on the object itself. There's an example of this https://www.baeldung.com/java-objects-hash-vs-objects-hashcode .

However it still would retain the property that any equal objects would have the same hashcode within the same JVM, which is sufficient. Across different JVMs (in a distributed context) we aren't guaranteed consistent hashcode for the same object. But this is an issue regardless of the method that's used https://martin.kleppmann.com/2012/06/18/java-hashcode-unsafe-for-distributed-systems.html


2.) That's a good point, although I was kind of against suppressing mostly because it seemed really unnecessary. In the case of more fields, I think it makes sense to just tackle that then. I'm not sure about if for these classes new fields would ever be added but of course, hard to predict that 

 Let me know what you think !

Reply:
Thanks for the explanation. I see this line in your blog:

```
The problem is that although the documentation says hashCode() doesn’t provide a consistency guarantee, the Java standard library behaves as if it did provide the guarantee. People start relying on it, and since backwards-compatibility is rated so highly in the Java community, it will probably never ever be changed, even though the documentation would allow it to be changed. So the JVM gets the worst of both worlds: a hash table implementation that is open to DoS attacks, but also a hash function that can’t always safely be used for communication between processes. :(

Therefore…

So what I’d like to ask for is this: if you’re building a distributed framework based on the JVM, please don’t use Java’s hashCode() for anything that needs to work across different processes. Because it’ll look like it works fine when you use it with strings and numbers, and then someday a brave soul will use (e.g.) a protocol buffers object, and then spend days banging their head against a wall trying to figure out why messages are getting sent to the wrong servers.
```

In that case, it feels like we should really update the errorprone config to not report warning for not using hashCode...

Reply:
Updated the error prone config!